server/src/test/java/org/opensearch/search/pipeline/SearchPipelineServiceTests.java

```diff
@@ -505,6 +569,93 @@ public void testTransformResponse() throws Exception {
         );
     }
 
+    public void testTransformSearchPhase() {
+        SearchPipelineService searchPipelineService = createWithProcessors();
+        SearchPipelineMetadata metadata = new SearchPipelineMetadata(
+            Map.of(
+                "p1",
+                new PipelineConfiguration(
+                    "p1",
+                    new BytesArray("{\"phase_injector_processors\" : [ { \"max_score\" : { } } ]}"),
+                    XContentType.JSON
+                )
+            )
+        );
+        ClusterState clusterState = ClusterState.builder(new ClusterName("_name")).build();
+        ClusterState previousState = clusterState;
+        clusterState = ClusterState.builder(clusterState)
+            .metadata(Metadata.builder().putCustom(SearchPipelineMetadata.TYPE, metadata))
+            .build();
+        searchPipelineService.applyClusterState(new ClusterChangedEvent("", clusterState, previousState));
+        SearchPhaseController controller = new SearchPhaseController(
+            writableRegistry(),
+            s -> InternalAggregationTestCase.emptyReduceContextBuilder()
+        );
+        SearchPhaseContext searchPhaseContext = new MockSearchPhaseContext(10);
+        QueryPhaseResultConsumer searchPhaseResults = new QueryPhaseResultConsumer(
+            searchPhaseContext.getRequest(),
+            OpenSearchExecutors.newDirectExecutorService(),
+            new NoopCircuitBreaker(CircuitBreaker.REQUEST),
+            controller,
+            SearchProgressListener.NOOP,
+            writableRegistry(),
+            2,
+            exc -> {}
+        );
+
+        final QuerySearchResult querySearchResult = new QuerySearchResult();
+        querySearchResult.setShardIndex(1);
+        querySearchResult.topDocs(new TopDocsAndMaxScore(new TopDocs(null, new ScoreDoc[1]), 100f), null);
```

Comment:
Is the input data already setting the max score to `100f`? If so, how do we know that the transformer ran (since it also sets the max score to `100f`)?

Reply:
will fix it